runtime/src/append_vec.rs

```diff
@@ -89,8 +89,8 @@ impl<'a> StoredAccount<'a> {
     }
 }
 
-#[derive(Debug)]
 #[allow(clippy::mutex_atomic)]
+#[derive(Debug, AbiSample)]
```

Comment:
Yeah, this reorder is needed to avoid the current proc's macro limitation. But this PR is already too big... I think this reorder is tolerable.